File: practice26/8.cpp


//https://www.nowcoder.com/practice/56c6fb8231a44ed8ab91ac231f7b2c63?tpId=98&tqId=32949&tPage=7&rp=7&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;
int main() {
	string str;
	cin >> str;
	int n, i;
	cin >> n;
	if (str.size() < n) {
		cout << -1 << endl;
	}
	else {
		for (i = 0; i < str.size() - n; i++) {
			cout << str.substr(i, n) << ' ';
		}
		cout << str.substr(i, n);
	}
	system("pause");
	return 0;
}

File: practice32/8.java



//https://www.nowcoder.com/practice/1183548cd48446b38da501e58d5944eb?tpId=85&tqId=29840&tPage=1&rp=1&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int col=sc.nextInt();
        int row=sc.nextInt();
        int[][] pt=new int[row][col];
        int count=0;
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(pt[i][j]==0){
                    count++;
                    if(i+2<row){
                        pt[i+2][j]=-1;
                    }
                    if(j+2<col){
                        pt[i][j+2]=-1;
                    }
                }
            }
        }
        System.out.println(count);
    }
}